services/storage/src/simcore_service_storage/dsm.py

```diff
@@ -472,29 +473,23 @@ async def try_update_database_from_storage(
 
                     return to_meta_data_extended(row)
         except botocore.exceptions.ClientError:
-            if silence_exception:
-                logger.debug("Error happened while trying to access %s", file_uuid)
-            else:
-                logger.warning(
-                    "Error happened while trying to access %s", file_uuid, exc_info=True
-                )
-            # the file is not existing or some error happened
-            return None
+            logger.debug("Error happened while trying to access %s", file_uuid)
+            if reraise_exceptions:
+                raise
 
-    @retry(
-        stop=stop_after_delay(1 * _HOUR),
-        wait=wait_exponential(multiplier=0.1, exp_base=1.2, max=30),
-        retry=(
-            retry_if_exception_type() | retry_if_result(lambda result: result is None)
-        ),
-        before_sleep=before_sleep_log(logger, logging.INFO),
-    )
     async def auto_update_database_from_storage_task(
         self, file_uuid: str, bucket_name: str, object_name: str
     ) -> Optional[FileMetaDataEx]:
-        return await self.try_update_database_from_storage(
-            file_uuid, bucket_name, object_name, silence_exception=True
-        )
+        async for attempt in AsyncRetrying(
```

Comment:
Why not the decorator?

Reply:
Because in the traceback it was not clear enough when the error was raised. This makes it more readable. The errors are easier to spot now.